app/models/url.rb

```diff
@@ -0,0 +1,21 @@
+require "uri"
+class Url < ActiveRecord::Base
+  validates :long, :uniqueness => true, format: { with: /\A#{URI::regexp}\z/}
+  before_save :convert
+
+  private
+
+  def convert
+     range = [*'0'..'9', *'a'..'z', *'A'..'Z']
+     unless self.short
+        short_url = Array.new(8){range.sample}.join
+         convert if Url.where(short: short_url).length > 0
```

Comment:
I admire your use of recursion, but this is a bad use of recursion. ;)

Try changing `unless self.short` on line 10 to `until self.short`